### PR TITLE
Cover photo upload and delete

### PR DESCRIPTION
**Pull request by sharathnair9999**
- Upload and Delete a cover photo from logged user's profile
- Sort Explore Posts by latest and trending posts

### CODE REVIEW

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/helpers/sortMethods.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const sortByLatest = (list, bool) => {
```
Comment (bajajcodes) [inline src/helpers/sortMethods.js:1]: Sugestion: Indexing and query can also be used, if fits use-case and requirements.

File: src/pages/UserProfile/components/EditProfileModal.jsx [hunk 2]
```diff
@@ -53,21 +55,69 @@ const EditProfileModal = ({ setShowModal, userInfo }) => {
...
   useEffect(() => {
     if (file) {
+      console.log("uploading dp");
       uploadFile(file, setIsValid, setUserDetails, true, "photoURL");
       setFile("");
     }
   }, [file]);
+  useEffect(() => {
+    if (coverFile) {
+      console.log("uploading cover");
```
Comment (shivanigangadharan) [inline src/pages/UserProfile/components/EditProfileModal.jsx]: can remove consoles here if not reqd

File: src/pages/UserProfile/components/EditProfileModal.jsx [hunk 3]
```diff
@@ -53,21 +55,69 @@ const EditProfileModal = ({ setShowModal, userInfo }) => {
...
+      uploadFile(coverFile, setIsValid, setUserDetails, true, "coverPhoto");
+      setCoverFile("");
+    }
+  }, [coverFile]);
 
   useEffect(() => {
     setUserDetails(userInfo);
   }, [userInfo]);
 
+  console.log(userDetails);
```
Comment (shivanigangadharan) [inline src/pages/UserProfile/components/EditProfileModal.jsx]: here too

**Review by shivanigangadharan — APPROVED** [review]
Code looks good, nice implementation of features.

File: src/pages/UserProfile/components/EditProfileModal.jsx (hunk 3 above)
Comment (sharathnair9999) [reply src/pages/UserProfile/components/EditProfileModal.jsx]: done

File: src/pages/UserProfile/UserProfile.jsx [hunk 4]
```diff
@@ -54,10 +54,19 @@ const UserProfile = () => {
               />
             </Modal>
           )}
+          <img
+            className="absolute w-full h-40 object-cover -left-0 -z-10"
+            src={
+              userInfo.coverPhoto
```
Comment (bajajcodes) [inline src/pages/UserProfile/UserProfile.jsx:60]: Optional chaining operator might help here.

**Review by bajajcodes — COMMENTED** [review]
- Consistency is maintained.
- Good use of modularity and resuability.
- Left two suggestions.

File: src/helpers/sortMethods.js (hunk 1 above)
Comment (sharathnair9999) [reply src/helpers/sortMethods.js:1]: yeah will do that.

File: src/pages/UserProfile/UserProfile.jsx (hunk 4 above)
Comment (sharathnair9999) [reply src/pages/UserProfile/UserProfile.jsx:60]: thanks for that will change right away

**Pull request merged**